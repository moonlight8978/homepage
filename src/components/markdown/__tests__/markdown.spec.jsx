import React from 'react'
import { render } from '@testing-library/react'

import Markdown from '../markdown'

const content = `# Introduction

This is an introduction section.

# 1 First section

## 1.1 First sub heading
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

### 1.1.1 First inner heading
The \`map\` function is so declarative.
![Image](https://example.com/example.png)

## 1.2 Second sub heading
This is a long paragraph, contains **bold text**, links, and many more.
[See more here](https://example.com)

# 2 Second section
> This is a blockquote
`

test('renders correctly', () => {
  const { container, getByText, getByTestId } = render(
    <Markdown content={content} />
  )

  const heading = getByText('1.2 Second sub heading')
  expect(heading).toBeInTheDocument()
  expect(heading.tagName).toBe('H2')

  const image = getByTestId('markdown-image')
  expect(image.getAttribute('src')).toBe('https://example.com/example.png')
  expect(image.getAttribute('alt')).toBe('Image')
  const imageCaption = getByText('Image')
  expect(imageCaption).toBeInTheDocument()

  const link = getByText('See more here')
  expect(link).toBeInTheDocument()
  expect(link.getAttribute('href')).toBe('https://example.com')

  const strong = getByText('bold text')
  expect(strong).toBeInTheDocument()
  expect(strong.tagName).toBe('STRONG')

  const code = getByText('map')
  expect(code).toBeInTheDocument()
  expect(code.tagName).toBe('CODE')

  const blockquote = getByTestId('markdown-blockquote')
  expect(blockquote.textContent).toBe('This is a blockquote')

  expect(container.firstChild).toMatchSnapshot()
})
