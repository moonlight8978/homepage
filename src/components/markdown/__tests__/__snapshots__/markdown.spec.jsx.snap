// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="markdownBody"
>
  <h1
    class="h1"
  >
    Introduction
  </h1>
  <div
    class="paragraph"
  >
    This is an introduction section.
  </div>
  <h1
    class="h1"
  >
    1 First section
  </h1>
  <h2
    class="h2"
  >
    1.1 First sub heading
  </h2>
  <div
    class="paragraph"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  </div>
  <h3
    class="h3"
  >
    1.1.1 First inner heading
  </h3>
  <div
    class="paragraph"
  >
    The 
    <code
      class="code"
    >
      map
    </code>
     function is so declarative.

    <figure
      class="container"
    >
      <img
        alt="Image"
        class="image"
        data-testid="markdown-image"
        src="https://example.com/example.png"
      />
      <figcaption
        class="caption"
      >
        Image
      </figcaption>
    </figure>
  </div>
  <h2
    class="h2"
  >
    1.2 Second sub heading
  </h2>
  <div
    class="paragraph"
  >
    This is a long paragraph, contains 
    <strong
      class="strong"
      data-testid="markdown-strong"
    >
      bold text
    </strong>
    , links, and many more.

    <a
      class="link"
      data-testid="markdown-link"
      href="https://example.com"
    >
      See more here
    </a>
  </div>
  <h1
    class="h1"
  >
    2 Second section
  </h1>
  <blockquote
    class="blockquote"
    data-testid="markdown-blockquote"
  >
    <div
      class="paragraph"
    >
      This is a blockquote
    </div>
  </blockquote>
  <h1
    class="h1"
  >
    3 Code sample
  </h1>
  <pre
    class="container"
  >
    <code
      class="language-javascript code"
    >
      import React from 'react'
    </code>
  </pre>
  <h1
    class="h1"
  >
    List demo
  </h1>
  <ul
    class="list"
  >
    <li>
      Try hard
    </li>
    <li>
      Try harder
    </li>
  </ul>
  <ol
    class="list"
  >
    <li>
      Prepare
    </li>
    <li>
      Go
    </li>
  </ol>
</div>
`;
